aws/resource_aws_elasticache_replication_group.go

```diff
@@ -223,6 +225,7 @@ func resourceAwsElasticacheReplicationGroup() *schema.Resource {
 			},
 		},
 		SchemaVersion: 1,
+		MigrateState:  resourceAwsElasticacheReplicationGroupMigrateState,
```

Comment:
Since this is an exceptional case and we want to perform the simplest possible upgrade for compatibility, can you please inline this with the following comments?

```suggestion
		// SchemaVersion: 1 did not include any state changes via MigrateState.
		// Perform a no-operation state upgrade for Terraform 0.12 compatibility.
		// Future state migrations should be performed with StateUpgraders.
		MigrateState: func(v int, inst *terraform.InstanceState, meta interface{}) (*terraform.InstanceState, error) {
			return inst, nil
		},
```

I verified this successfully with a Terraform AWS Provider 1.13.0 `aws_elasticache_replication_group` resource on Terraform 0.12.1. 👍

Reply:
Sure, I'll swap over. I do find the replacement more error prone, but either way is fine by me.

Reply:
Hi @akatrevorjay, the passthrough implementation is not more error prone -- `MigrateState` and dealing with flatmap state migrations in of the Terraform v0.12's Provider SDK has been superseded with JSON state migrations via `StateUpgraders`. In this case, we do not need to worry about doing anything other than passing through the existing `InstanceState` to bypass the Terraform runtime error, which matches the Terraform v0.11 Provider SDK in this situation where `MigrateState` was not implemented alongside a `SchemaVersion` update. We'll be removing the upstream runtime error in the next version of Terraform v0.12 (https://github.com/hashicorp/terraform/pull/21625) to match the old Terraform v0.11 behavior.

Reply:
Aha! I didn't see any mention of `StateUpgraders` in the docs at the time of writing this, maybe missed it! Thanks so much for the explanation!